yldio/asap-hub
# Upgrade react-scripts

Current work:
By upgrading to React Scripts 5.01 we are upgrading major versions of webpack 5, Jest 27, and ESLint 8. This will unblock our upgrade path for some of our core dependencies most notable React itself!

Deployment:
Changes to webpack have altered our auth0 login page HTML. I think the old HTML will work but due do auth0 having shared envs is not easy to test and can only be done in dev/prod. We will need a maintenance window to deploy these changes.

Testing:
We should try and regression test the entire website as much as possible. Some areas I have had to do more migration work are:
- Teams, user and research output exports
	- Client might need to test this as there were problems with the initial implementation and windows excel. I have tested with mac Excel and it seems okay.
	- I can't find the buttons to export.
- Login pages (these can only be tested properly in dev)
- Rich text format
- GP2
- Avatar upload

Future Work
I will bring this up in RTM today discuss priority and impact  of these issues. I think they are all pretty important.

Upgrade Typescript

Change FE build output to ESM and drop building CJS versions. (https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c)
- Once we are outputting ESM we can update Unified to Version 10 and remove the forced Resolution for VFile
- Speed up pipeline by no-longer compiling for CJS
- Remove jest hack for csv-stream ('^csv-stringify/browser/esm': require.resolve('csv-stringify'),)

Upgrade React to 18 

Speed up pipeline with Jest sharding

Fix the problem with Broken timers on Event tests
- I am going to comment this test out and ignore coverage complaints. 

Investigate the bundle file size
- Investigate the main.js bundle size ballooning to 260kb+ (from 160kb which was already bumped from 128 kb max bundle size limit we had set before)
- Auth Frontend looks to have hit 900kb
	- This doesn't look related to react scripts changes it looks like its been a problem for a while
